From d1e94f8bc4f7434c6793d12614d88d3770093f71 Mon Sep 17 00:00:00 2001
From: Tri Vo <trong@google.com>
Date: Tue, 18 Jul 2023 14:53:39 -0400
Subject: [PATCH 1/2] Keystore: Use waitForService() instead of getService()

getService() is deprecated. It can also time out, and keystore doesn't
currently deal with that error condition gracefully.

Bug: 285846920
Test: boot device, IServiceManager doesn't complain
Change-Id: I3b73bf29087cc452d57545a5fd2aa9a3a50a86d4
(cherry picked from commit 8d08cd9c5e146ab0fc868385e68a4c6547e5b996)
---
 keystore/keystore_attestation_id.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/keystore/keystore_attestation_id.cpp b/keystore/keystore_attestation_id.cpp
index ccd38085..75c62dd7 100644
--- a/keystore/keystore_attestation_id.cpp
+++ b/keystore/keystore_attestation_id.cpp
@@ -74,8 +74,8 @@ KeyAttestationApplicationIdProvider& KeyAttestationApplicationIdProvider::get()
 }
 
 KeyAttestationApplicationIdProvider::KeyAttestationApplicationIdProvider()
-    : BpKeyAttestationApplicationIdProvider(
-          android::defaultServiceManager()->getService(String16("sec_key_att_app_id_provider"))) {}
+    : BpKeyAttestationApplicationIdProvider(android::defaultServiceManager()->waitForService(
+          String16("sec_key_att_app_id_provider"))) {}
 
 DECLARE_STACK_OF(ASN1_OCTET_STRING);
 
-- 
2.43.0

